docs/usage/rtk-query/cached-data.md

```diff
@@ -0,0 +1,747 @@
+```
+
+```tsx title="App.tsx"
+function App() {
+  const { data: posts } = useGetPostsQuery()
+  const [addPost] = useAddPostMutation()
+
+  return (
+    <div>
+      <AddPost onAdd={addPost} />
+      <PostsList />
+      {/* Assume each PostDetail is subscribed via `const {data} = useGetPostQuery(id)` */}
+      <PostDetail id={1} />
+      <PostDetail id={2} />
+      <PostDetail id={3} />
+    </div>
+  )
+}
+```
+
+**What to expect**
+
+When `addPost` is triggered, it would cause each `PostDetail` component to go back into a `isFetching` state because `addPost` invalidates the root tag, which causes _every query_ that provides 'Posts' to be re-run. In most cases, this may not be what you want to do. Imagine if you had 100 posts on the screen that all subscribed to a `getPost` query – in this case, you'd create 100 requests and send a ton of unnecessary traffic to your server, which we're trying to avoid in the first place! Even though the user would still see the last good cached result and potentially not notice anything other than their browser hiccuping, you still want to avoid this.
+
+#### Selectively invalidating lists
+
+```ts title="API Definition"
+export const api = createApi({
+  baseQuery: fetchBaseQuery({ baseUrl: '/' }),
+  tagTypes: ['Posts'],
+  endpoints: (build) => ({
+    getPosts: build.query<PostsResponse, void>({
+      query: () => 'posts',
+      providesTags: (result) =>
+        result
+          ? [
+              ...result.map(({ id }) => ({ type: 'Posts', id })),
+              { type: 'Posts', id: 'LIST' },
+            ]
+          : [{ type: 'Posts', id: 'LIST' }],
+    }),
+    addPost: build.mutation<Post, Partial<Post>>({
+      query(body) {
+        return {
+          url: `posts`,
+          method: 'POST',
+          body,
+        }
+      },
+      invalidatesTags: [{ type: 'Posts', id: 'LIST' }],
+    }),
+    getPost: build.query<Post, number>({
+      query: (id) => `posts/${id}`,
+      providesTags: (result, error, id) => [{ type: 'Posts', id }],
+    }),
+  }),
+})
+
+export const { useGetPostsQuery, useAddPostMutation, useGetPostQuery } = api
+```
+
+```tsx title="App.tsx"
+function App() {
+  const { data: posts } = useGetPostsQuery()
+  const [addPost] = useAddPostMutation()
+
+  return (
+    <div>
+      <AddPost onAdd={addPost} />
+      <PostsList />
+      {/* Assume each PostDetail is subscribed via `const {data} = useGetPostQuery(id)` */}
+      <PostDetail id={1} />
+      <PostDetail id={2} />
+      <PostDetail id={3} />
+    </div>
+  )
+}
+```
+
+**What to expect**
+
+When `addPost` is fired, it will only cause the `PostsList` to go into an `isFetching` state because `addPost` only invalidates the 'LIST' id, which causes `getPosts` to rerun (because it provides that specific id). So in your network tab, you would only see 1 new request fire for `GET /posts`. As the singular `getPost` queries have not been invalidated, they will not re-run as a result of `addPost`.
```

Comment:
```suggestion
When `addPost` is fired, it will only cause the `PostsList` to go into an `isFetching` state because `addPost` only invalidates the `'LIST'` id, which causes `getPosts` to rerun (because it provides that specific id). So in your network tab, you would only see 1 new request fire for `GET /posts`. As the singular `getPost` queries have not been invalidated, they will not re-run as a result of `addPost`.
```